Cabal/Distribution/Package.hs

```diff
@@ -77,12 +116,16 @@ import Text.PrettyPrint ((<+>), text)
 -- This type is opaque since @Cabal-2.0@
 --
 -- @since 2.0
-newtype PackageName = PackageName ShortText
-    deriving (Generic, Read, Show, Eq, Ord, Typeable, Data)
+newtype PackageName = PackageName UnqualComponentName
```

Comment:
I'm.... not sure setting it up this way makes sense? A package name isn't really a component name in any sense; they just happen to coincide in syntax due to some historical choices. Better to keep them disjoint? (If the duplication of the Text instance is troublesome, define a helper function.)